javascript/ql/lib/semmle/javascript/frameworks/CryptoLibraries.qll

```diff
@@ -158,11 +169,25 @@ private module NodeJSCrypto {
       exists(DataFlow::SourceNode mod |
         mod = DataFlow::moduleImport("crypto") and
         this = mod.getAMemberCall("create" + ["Hash", "Hmac", "Sign", "Cipher"]) and
-        algorithm.matchesName(this.getArgument(0).getStringValue())
+        algorithmName = this.getArgument(0).getStringValue() and
+        algorithm.matchesName(algorithmName)
```

Comment:
Instead of having both `algorithm` and `algorithmName` as fields, you could have just `algorithmName` as a field.  
Then the implementation for `getAlgorithm()` would become `result.mathcesName(algorithmName)`.

Reply:
👍